compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -3173,13 +3338,33 @@ public BLangNode currentAction() {
             return this.actions.get(this.currentState);
         }
 
+        private boolean checkAndSetWorkerMultipleReceiveSource(BLangWorkerMultipleReceive multipleReceive,
+                                                               String sourceWorkerId) {
+
+            for (WorkerMultipleReceiveNode.WorkerReceiveFieldNode receivedWorker : multipleReceive.receiveFields) {
+                if (!receivedWorker.getHasReceived()
+                        && receivedWorker.getWorkerName().getValue().equals(sourceWorkerId)) {
+                    multipleReceive.currentReceiveFieldIndex = multipleReceive.receiveFields.indexOf(receivedWorker);
+                    receivedWorker.setHasReceived(true);
+                    return true;
+                }
+            }
+            return false;
+        }
+
         public boolean currentIsReceive(String sourceWorkerId) {
+
             if (this.done()) {
                 return false;
             }
             BLangNode action = this.currentAction();
-            return !isWorkerSend(action) && !isWorkerSyncSend(action) &&
-                    ((BLangWorkerReceive) action).workerIdentifier.value.equals(sourceWorkerId);
+            if (isWorkerMultipleReceive(action)) {
```

Comment:
Shall we take result of `!isWorkerSend(action) && !isWorkerSyncSend(action)` to a variable before the if condition.

Reply:
Fixed